### ISSUE TITLE
[Bug] Map initialization crashing

### ISSUE THREAD

**Issue by kevinjmo** [visgl/react-map-gl#2221]
### Description

After the map attempts to load, I’m receiving these errors

```
TypeError: undefined is not an object (evaluating 'this.transform’)

The above error occurred in the <ForwardRef(Map)> component:
```

I passed a callback that logs when the map is loaded via `onLoad` and that statement gets rendered just before these errors throw. 

### Steps to Reproduce

We have been using `react-map-gl` for a year without any issue. However, it seems like the recent update broke something.

### Environment

- Framework version:react-map-gl@7.1.1
- Map library: mapbox-gl@1.13.3
- Browser: Safari 16.5.1

### Logs

```
TypeError: undefined is not an object (evaluating 'this.transform’)

The above error occurred in the <ForwardRef(Map)> component:
```

**Comment by Pessimistress** [comment]
Please provide a code sandbox that reproduces your issue.

**Comment by kevinjmo** [comment]
I’m developing on in an enterprise environment at work so I’m not able to recreate it in CodeSandbox.

However, I figured to post in case someone has come across this or someone might know what’s going on off the top of their head ¯\_(ツ)_/¯

**Comment by Pessimistress** [comment]
Well, what was the last version that worked? Which line did the error originate from? There isn't even a `this.transform` in the current code base. Unless you can share more information I'll have to close this bug as not reproducible. You can always use Discussions if you just want to pick other users brains.

**Comment by kevinjmo** [comment]
Thank you for taking the time, Xiaoji, I really appreciate it. I was digging deeper and noticed it was coming from this stack trace

```
TypeError: undefined is not an object (evaluating 'this.transform')
(anonymous function) — mapbox-gl.js:35
_updateSettings — mapbox.js:382
setProps — mapbox.js:138
(anonymous function) — map.js:78
```

Following this stack trace…

Inspecting `(anonymous function) — map.js:78` in the browser, it leads us to this line in map.tsx

https://github.com/visgl/react-map-gl/blob/2c8951c93ba734b38b85d99a6ea5ed38d6e470d0/src/components/map.tsx#L114

`setProps — mapbox.js:138` leads us to

https://github.com/visgl/react-map-gl/blob/2c8951c93ba734b38b85d99a6ea5ed38d6e470d0/src/mapbox/mapbox.ts#L256


`(anonymous function) — mapbox-gl.js:35` leads to 

```js
                setter === null || setter === void 0 ? void 0 : setter(nextProps[propName]);
```

Does this help at all?

**Comment by kevinjmo** [comment]
Upon debugging in the browser, it looks like it was due to setting `maxZoom` and `minZoom` on the Map component. Remove those two props on the Map component solved it from throwing, however, it’s a bit strange as those two props were working fine before.

<img width="607" alt="Screenshot 2023-07-06 at 13 54 59" src="https://github.com/visgl/react-map-gl/assets/9851651/41c28c7f-5dbf-44fb-a8ce-a2edd8ace3a7">

**Comment by Pessimistress** [comment]
Thanks, that is very helpful. I am able to reproduce the error on my end.

**Comment by kevinjmo** [comment]
Great! If it helps, I noticed that setting `maxZoom` or `minZoom` to any primitive values seems to work fine. However, when conditionally setting it like so, it seems to throw the error
```jsx
// Does not work
<ReactMap
  maxZoom={defaultLayer?.maxzoom || 22}
  minZoom={defaultLayer?.minzoom || 2}

**Comment by Pessimistress** [comment]
Published v7.1.2.

**Comment by NitayMG** [comment]
still have this issue,
import React, { useRef, useEffect, useState, useContext, useCallback} from 'react';;
import Map, { Marker, Popup, NavigationControl, ScaleControl, } from 'react-map-gl';
import 'mapbox-gl/dist/mapbox-gl.css';

export default function Test() {

  const [mapType, setMapType] = useState(process.env.REACT_APP_MAPBOX_BASIC_MAP);
  const [showMarkers, setShowMarkers] = useState(true);
  const [zoom, setZoom] = useState(50);
  const [center, setCenter] = useState( ({lat: 32.096234, lng: 34.847722}));
  const [devicesArray, setDevicesArray] = useState([]);
  const [devicesImeiArray, setDevicesImeiArray] = useState([]);
  const[latitudeAverage, setLatitudeAverage] = useState(0);
  const[longitudeAverage, setLongitudeAverage] = useState(0);
  
     const initialRegionByGroup = (devicesArray) => { 
      let latSum = 0, lngSum = 0, counter = 0;
      if(devicesArray.length > 0){
        for (let i = 0; i < devicesArray.length; i++) {
          latSum += parseFloat(devicesArray[i].data.lat);//converting to numbers
          lngSum += parseFloat(devicesArray[i].data.lng);//converting to numbers
          counter++;
        }
        latSum = latSum/counter;
        lngSum = lngSum/counter;
        }
        if(lngSum - latSum < 13 && latSum - lngSum < 13){//if the distance between the longitude to the latitude is small so initial the map to it
          setLatitudeAverage(latSum);
          setLongitudeAverage(lngSum);
          // console.log(`Success, the distance between the longitude to the latitude is small`)
        }
        else{//if the distance between the longitude to the latitude is too big so initial the map to it
          setLatitudeAverage(32.096234);
          setLongitudeAverage(34.847722);
          // console.log(`Fail, the distance between the longitude to the latitude is too big`)
        }
      };

      useEffect(() => {
        initialRegionByGroup(devicesArray);
      }, [devicesArray]);
      
    //This function change the type of the after clicking in the image botton "MapLayer"
  const changeMapType = () => {
    const mapTypes = [
      process.env.REACT_APP_MAPBOX_BASIC_MAP,
      process.env.REACT_APP_MAPBOX_OUTDOORS_3D_MAP,
      process.env.REACT_APP_MAPBOX_WHITE_MAP_3D,
      process.env.REACT_APP_MAPBOX_NIGHT_MAP,
      process.env.REACT_APP_MAPBOX_MOUNTAINS_MAP,
      process.env.REACT_APP_MAPBOX_SATELLITE_MAP_3D,
      process.env.REACT_APP_MAPBOX_SATELLITE_MAP_3D_WITH_3D_BUILDINGS
    ];
    const currentMapTypeIndex = mapTypes.indexOf(mapType);
    const nextMapTypeIndex = (currentMapTypeIndex + 1) % mapTypes.length;
    const nextMapType = mapTypes[nextMapTypeIndex];
    setTimeout(() => {
      setMapType(nextMapType);
    }, 500);
  };

    // Zoom out to view the entire group by a function that calculates an average of positions every few seconds
    const zoomOutToGroup = () => {
      setCenter({lat:latitudeAverage,lng:longitudeAverage});
      setZoom(8);
    };
    
    //This function set the view of the map to focus on the marker we clicked
    const handleClickActiveMarker = (marker) => {  
      setCenter({lng:JSON.parse(marker.lng), lat:JSON.parse(marker.lat)}); // set the center of the map to be the position of the clicked marker
      setZoom(18);
    };

    

    
    const handleCenterChanged = (e) =>{
       // console.log(`CURRENT ZOOM: ${e.viewState}`)//this refers to Google Map instance
       setCenter({lat:e.viewState.latitude, lng:e.viewState.longitude});
    };

     const handleZoomChanged = (e) =>{
       // console.log(`CURRENT ZOOM: ${e.viewState.zoom}`)//this refers to Google Map instance
       setZoom(e.viewState.zoom);
    };

  return (
    <>
    <Map
    
      mapboxAccessToken={process.env.REACT_APP_MAPBOX_ACCESS_TOKEN} 
      // This cordiantes happen only on the first load of map
      initialViewState={{
        longitude: center.lng ?? 34.847722,
        latitude: center.lat ?? 32.096234,
        zoom: zoom ?? 6,
        }}
      longitude={center.lng ?? 34.847722}
      latitude={center.lat ?? 32.096234}
      zoom={zoom ?? 6}
      onMove={(e) => handleCenterChanged(e)}
      onZoom={(e) => handleZoomChanged(e)}
      attributionControl={false}
      style={{ height: "80vh", width: "100%" }}
      mapStyle={process.env.REACT_APP_MAPBOX_MAPS_API_STYLE+`${mapType}`}
      >
      
    {showMarkers &&
      devicesArray?.map((device, index) =>(
        <React.Fragment key={device.config.url.substring(device.config.url.length - 15)}>
          <Marker longitude={parseFloat(device.data.lng) || 0} latitude={parseFloat(device.data.lat) || 0} >
            <div className="marker" onClick={() => {openPopup(index);handleClickActiveMarker(device.data);}}>
              <img src={`./Images/${MarkerColor(device)}.png`} alt="marker icon" />
            </div>
          </Marker>

        {selectedMarker === index && (
            <Popup
             key={device.config.url.substring(device.config.url.length - 15)}
             longitude={device.data.lng}
             latitude={device.data.lat}
             onClose={closePopup}
             closeButton={true}
             closeOnClick={false}
             offsetTop={-30}
            > 
               <MarkerInfoWindowData device={device}/>
           </Popup>
          )}
        </React.Fragment>
      ))}

        <NavigationControl
         showCompass={true}
         showZoom={true}
         visualizePitch={true}
         />
         <ScaleControl position='bottom-right'/>
      </Map>
      </div>
    </div>
  </>
  )
};
